pkg/kubelet/dockertools/manager.go

```diff
@@ -1893,7 +1894,10 @@ func (dm *DockerManager) SyncPod(pod *api.Pod, runningPod kubecontainer.Pod, pod
 
 		// Find the pod IP after starting the infra container in order to expose
 		// it safely via the downward API without a race and be able to use podIP in kubelet-managed /etc/hosts file.
-		pod.Status.PodIP = dm.determineContainerIP(pod.Name, pod.Namespace, podInfraContainer)
+		pod.Status.PodIP, err = dm.determineContainerIP(pod.Name, pod.Namespace, podInfraContainer)
+		if err != nil {
+			return err
```

Comment:
Do we need to kill the infra container here? Why when SetUpPod() goes wrong, we kill it; but when InspectContainer goes wrong we doesn't kill it? What about here?